src/main/java/com/algorand/algosdk/builder/transaction/ApplicationCreateTransactionBuilder.java

```diff
@@ -0,0 +1,74 @@
+package com.algorand.algosdk.builder.transaction;
+
+import com.algorand.algosdk.logic.StateSchema;
+import com.algorand.algosdk.transaction.Transaction;
+
+@SuppressWarnings("unchecked")
+public class ApplicationCreateTransactionBuilder<T extends  ApplicationCreateTransactionBuilder<T>> extends ApplicationUpdateTransactionBuilder<T> {
+    private StateSchema localStateSchema;
+    private StateSchema globalStateSchema;
+
+    /**
+     * Initialize a {@link ApplicationCreateTransactionBuilder}.
+     */
+    public static ApplicationCreateTransactionBuilder<?> Builder() {
+        return new ApplicationCreateTransactionBuilder<>();
+    }
+
+    public ApplicationCreateTransactionBuilder() {
+        super.onCompletion(Transaction.OnCompletion.NoOpOC);
+        super.applicationId(0L);
+    }
+
+    @Override
+    protected void applyTo(Transaction txn) {
+        txn.localStateSchema = localStateSchema;
+        txn.globalStateSchema = globalStateSchema;
+
+        super.applyTo(txn);
+    }
+
+    /**
+     * This option is disabled for application create, where the ID must be changed from 0.
+     */
+    @Override
+    public T applicationId(Long appId) {
```

Comment:
Why do we need this method? Better to do away with it.

Reply:
I think I understand now.
It can also go to all the leaf classes, and not the base class, and avoid it here. 
I think it is okay as is.

Reply:
Yes thats right. The supported fields hierarchy didn't work out quite right, so I either "disable" the method here or have an extra base-class containing everything except this method.